eng/targets/ResolveReferences.targets

```
@@ -226,9 +226,15 @@
       <PackageReference Include="Microsoft.Internal.Extensions.Refs" Version="$(MicrosoftInternalExtensionsRefsPackageVersion)" PrivateAssets="All" />
     </ItemGroup>
 
-    <ItemGroup>
+    <ItemGroup Condition="'$(IsTestProject)' == 'true' OR '$(IsTestAssetProject)' == 'true' OR '$(IsBenchmarkProject)' == 'true' OR '$(IsSampleProject)' == 'true'">
+      <PackageReference Include="@(_LatestPackageReferenceWithVersion)" />
+    </ItemGroup>
+
+    <ItemGroup Condition="'$(IsTestProject)' != 'true' AND '$(IsTestAssetProject)' != 'true' AND '$(IsBenchmarkProject)' != 'true' AND '$(IsSampleProject)' != 'true'">
```

curious if we can just put this condition into IsImplicitlyDefined instead of the two item groups.

---

I think you can have a condition on an item or an itemgroup, but not on metadata, so either way we'd need two instances of PackageReference. I could turn that big condition into a property, though.

---

You can conditionalize metadata e.g. with an additional simplification to use `$(IsImplementationProject)` because that already excludes what you want to exclude
```suggestion
    <ItemGroup>
      <PackageReference Include="@(_LatestPackageReferenceWithVersion)">
        <IsImplicitlyDefined Condition=" '$(IsImplementationProject)' == '' ">true</ IsImplicitlyDefined>
       </PackageReference>
    </ItemGroup>
```